internal/uvm/vsmb.go

```diff
@@ -19,15 +21,15 @@ type VSMBShare struct {
 	vm           *UtilityVM
 	HostPath     string
 	refCount     uint32
-	name         string
-	allowedFiles []string
-	guestPath    string
-	readOnly     bool
+	Name         string
+	AllowedFiles []string
+	GuestPath    string
+	Options      hcsschema.VirtualSmbShareOptions
```

Comment:
Why do we need to export these?

Reply:
Gob encode only encodes exported fields and in previous iteration we used to encode the entire VSMBShare struct. No that we individually encode each fields exporting it might not be necessary. I will revert it if encoding/decoding works well without exporting them.